optuna/experimental_warning.py

```diff
@@ -0,0 +1,42 @@
+import warnings
+
+
+# Reference: https://github.com/cupy/cupy/blob/8d923719ced31fdff4a42702d395ec87a7e98045/cupy/util.pyx#L89  # NOQA
+def experimental(api_name: str) -> None:
```

Comment:
Can we name this function as a verb? I'm don't think that the logic of this function will change that much over time so we could be more specific. I'd personally rename this submodule and define it at e.g. `optuna.experimental.warn`.